posthog/queries/event_query/event_query.py

```diff
@@ -141,7 +151,9 @@ def _does_cohort_need_persons(self, prop: Property) -> bool:
         return False
 
     @cached_property
-    def _person_query(self):
+    def _person_query(self) -> PersonQuery:
+        if isinstance(self._filter, PropertiesTimelineFilter):
```

Comment:
This feels clunky in that we are inheriting but not really. This mega-class just got more exceptional than anywhere else.

Could we create a PropertiesTimelineEventQuery and override these functions to raise there instead of hoisting its concerns elsewhere?

Reply:
This is just because of typing, there _is_ a `PropertiesTimelineEventQuery`, the only problem remaining at `EventQuery` level is that its `filter` needs to include `PropertiesTimelineFilter` as one of its union members, because that attribute isn't a generic (and Python generics are lackluster). So I'd prefer not to have this guard, but the guard is simpler than making `filter` generic